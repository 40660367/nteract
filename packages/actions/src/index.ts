/**
 * @module actions
 */

import * as ActionTypes from "./actionTypes/index";

console.log("actiontypes in actions/actiontypes", ActionTypes);

export * from "./actions/index";
export * from "./actionTypes/index";
