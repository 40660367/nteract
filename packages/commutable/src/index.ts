/**
 * @module commutable
 */
// .....................................
// API Exports
//

export * from "./primitives";
export * from "./structures";
export * from "./outputs";
export * from "./cells";
export * from "./notebook";

/*
// from structures
export {
  emptyCodeCell,
  emptyMarkdownCell,
  emptyNotebook,
  monocellNotebook,
  createCodeCell,
  insertCellAt,
  insertCellAfter,
  deleteCell,
  appendCellToNotebook
} from "./structures";

// v4
export { StreamOutput, Output, createImmutableOutput } from "./v4";

export {
  createImmutableMimeBundle,
  makeDisplayData,
  makeErrorOutput,
  makeStreamOutput,
  makeExecuteResult,
  MimeBundle
} from "./outputs";

export {
  makeRawCell,
  makeCodeCell,
  makeMarkdownCell,
  ImmutableCodeCell,
  ImmutableMarkdownCell,
  ImmutableRawCell,
  ImmutableCell,
  CellType
} from "./cells";

export {
  toJS,
  stringifyNotebook,
  fromJS,
  parseNotebook,
  makeNotebookRecord,
  Notebook,
  ImmutableNotebook
} from "./notebook";

*/
