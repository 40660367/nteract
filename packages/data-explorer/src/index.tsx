// Vendor modules
import * as React from "react";

// Local modules
import DataResourceTransformGrid from "./charts/grid";
import { semioticSettings } from "./charts/settings";
import { MetadataWarning } from "./components/MetadataWarning";
import { Toolbar } from "./components/Toolbar";
import { colors } from "./settings";
import { FlexItem, FlexWrapper, SemioticWrapper } from "./styled";
import * as Dx from "./types";
import { generateChartKey } from "./utilities";
import VizControls from "./VizControls";

// Constants
const mediaType = "application/vnd.dataresource+json";
const defaultResponsiveSize = [500, 300];
const defaultHeight = 500;

interface Props {
  /**
   * A pandas data object
   * e.g.
   * ```typescript
   * {
   *  shema: {},
   *  data: []
   * }
   * ```
   */
  data: Dx.DataProps;
  /**
   * An object that represents the current state of the Data Explorer.
   * e.g.
   * ```typescript
   * ```
   */
  metadata: Dx.Metadata;
  /**
   *
   */
  theme?: string;
  /**
   *
   */
  expanded?: boolean;
  /**
   *
   */
  height?: number;
  /**
   *
   */
  mediaType: "application/vnd.dataresource+json";
  /**
   *
   */
  initialView: Dx.View;
  /**
   *
   */
  onMetadataChange?: (
    { dx }: { dx: Dx.DxMetaProps },
    mediaType: string
  ) => void;
}

interface State {
  view: Dx.View;
  colors: string[];
  metrics: Dx.Field[];
  dimensions: Dx.Dimension[];
  selectedMetrics: string[];
  selectedDimensions: string[];
  networkType: Dx.NetworkType;
  hierarchyType: Dx.HierarchyType;
  pieceType: Dx.PieceType;
  summaryType: Dx.SummaryType;
  lineType: Dx.LineType;
  areaType: Dx.AreaType;
  chart: Dx.Chart;
  displayChart: Dx.DisplayChart;
  primaryKey: string[];
  data: Dx.Datapoint[];
}

class DataExplorer extends React.PureComponent<Partial<Props>, State> {
  static MIMETYPE: string = mediaType;

  static defaultProps: Partial<Props> = {
    metadata: {
      dx: {}
    },
    height: defaultHeight,
    mediaType,
    initialView: "grid"
  };

  constructor(props: Props) {
    super(props);

    const { metadata, initialView } = props;

    // Handle case of metadata being empty yet dx not set
    const dx = metadata.dx || {};
    const chart = dx.chart || {};

    const { fields = [], primaryKey = [] } = props.data.schema;

    const dimensions = fields.filter(
      field =>
        field.type === "string" ||
        field.type === "boolean" ||
        field.type === "datetime"
    ) as Dx.Dimension[];

    // Should datetime data types be transformed into js dates before
    // getting to this resource?
    const data = props.data.data.map(datapoint => {
      const mappedDatapoint: Dx.Datapoint = {
        ...datapoint
      };
      fields.forEach(field => {
        if (field.type === "datetime") {
          mappedDatapoint[field.name] = new Date(mappedDatapoint[field.name]);
        }
      });
      return mappedDatapoint;
    });

    const metrics = fields
      .filter(
        field =>
          field.type === "integer" ||
          field.type === "number" ||
          field.type === "datetime"
      )
      .filter(
        field => !primaryKey.find(pkey => pkey === field.name)
      ) as Dx.Metric[];

    const displayChart: Dx.DisplayChart = {};

    this.state = {
      view: initialView,
      lineType: "line",
      areaType: "hexbin",
      selectedDimensions: [],
      selectedMetrics: [],
      pieceType: "bar",
      summaryType: "violin",
      networkType: "force",
      hierarchyType: "dendrogram",
      dimensions,
      metrics,
      colors,
      chart: {
        metric1: (metrics[0] && metrics[0].name) || "none",
        metric2: (metrics[1] && metrics[1].name) || "none",
        metric3: "none",
        dim1: (dimensions[0] && dimensions[0].name) || "none",
        dim2: (dimensions[1] && dimensions[1].name) || "none",
        dim3: "none",
        timeseriesSort: "array-order",
        networkLabel: "none",
        ...chart
      },
      displayChart,
      primaryKey,
      data,
      ...dx
    };
  }

  componentDidMount(): void {
    // This is necessary to render any charts based on passed metadata
    // because the grid doesn't result from the updateChart function
    // but any other view does
    if (this.state.view !== "grid") {
      this.updateChart(this.state);
    }
  }

  updateChart = (updatedState: Partial<State>) => {
    const {
      view,
      dimensions,
      metrics,
      chart,
      lineType,
      areaType,
      selectedDimensions,
      selectedMetrics,
      pieceType,
      summaryType,
      networkType,
      hierarchyType,
      colors,
      primaryKey,
      data: stateData
    } = { ...this.state, ...updatedState };

    if (!this.props.data && !this.props.metadata && !this.props.initialView) {
      return;
    }

    const { data, height, onMetadataChange } = this.props;

    const { Frame, chartGenerator } = semioticSettings[view];

    const chartKey = generateChartKey({
      view,
      lineType,
      areaType,
      selectedDimensions,
      selectedMetrics,
      pieceType,
      summaryType,
      networkType,
      hierarchyType,
      chart
    });

    const frameSettings = chartGenerator(stateData, data!.schema, {
      metrics,
      dimensions,
      chart,
      colors,
      height,
      lineType,
      areaType,
      selectedDimensions,
      selectedMetrics,
      pieceType,
      summaryType,
      networkType,
      hierarchyType,
      primaryKey,
      setColor: this.setColor
    });

    const display: React.ReactNode = (
      <SemioticWrapper>
        <Frame
          responsiveWidth
          size={defaultResponsiveSize}
          {...frameSettings}
        />
        <VizControls
          {...{
            data: stateData,
            view,
            chart,
            metrics,
            dimensions,
            selectedDimensions,
            selectedMetrics,
            hierarchyType,
            summaryType,
            networkType,
            updateChart: this.updateChart,
            updateDimensions: this.updateDimensions,
            setLineType: this.setLineType,
            updateMetrics: this.updateMetrics,
            lineType,
            setAreaType: this.setAreaType,
            areaType
          }}
        />
      </SemioticWrapper>
    );

    // If you pass an onMetadataChange function, then fire it and
    // pass the updated dx settings so someone upstream can update
    // the metadata or otherwise use it
    if (onMetadataChange) {
      onMetadataChange(
        {
          ...this.props.metadata,
          dx: {
            view,
            lineType,
            areaType,
            selectedDimensions,
            selectedMetrics,
            pieceType,
            summaryType,
            networkType,
            hierarchyType,
            colors,
            chart
          }
        },
        mediaType
      );
    }

    this.setState(
      (prevState): any => {
        return {
          ...updatedState,
          displayChart: {
            ...prevState.displayChart,
            [chartKey]: display
          }
        };
      }
    );
  };

  setView = (view: Dx.View) => this.updateChart({ view });

  setGrid = () => this.setState({ view: "grid" });

  setColor = (newColorArray: string[]) =>
    this.updateChart({ colors: newColorArray });

  setLineType = (selectedLineType: Dx.LineType) =>
    this.updateChart({ lineType: selectedLineType });

  setAreaType = (selectedAreaType: Dx.AreaType) =>
    this.updateChart({ areaType: selectedAreaType });

  updateDimensions = (selectedDimension: string) => {
    const oldDims = this.state.selectedDimensions;
    const newDimensions =
      oldDims.indexOf(selectedDimension) === -1
        ? [...oldDims, selectedDimension]
        : oldDims.filter(dimension => dimension !== selectedDimension);
    this.updateChart({ selectedDimensions: newDimensions });
  };

  updateMetrics = (selectedMetric: string) => {
    const oldMetrics = this.state.selectedMetrics;
    const newMetrics =
      oldMetrics.indexOf(selectedMetric) === -1
        ? [...oldMetrics, selectedMetric]
        : oldMetrics.filter(metric => metric !== selectedMetric);
    this.updateChart({ selectedMetrics: newMetrics });
  };

  render(): JSX.Element {
    const {
      view,
      dimensions,
      chart,
      lineType,
      areaType,
      selectedDimensions,
      selectedMetrics,
      pieceType,
      summaryType,
      networkType,
      hierarchyType
    } = this.state;

    let display: React.ReactNode = null;

    if (view === "grid") {
      display = <DataResourceTransformGrid {...this.props as Props} />;
    } else if (
      [
        "line",
        "scatter",
        "bar",
        "network",
        "summary",
        "hierarchy",
        "hexbin",
        "parallel"
      ].includes(view)
    ) {
      const chartKey = generateChartKey({
        view,
        lineType,
        areaType,
        selectedDimensions,
        selectedMetrics,
        pieceType,
        summaryType,
        networkType,
        hierarchyType,
        chart
      });

      display = this.state.displayChart[chartKey];
    }

    return (
      <div>
        <MetadataWarning metadata={this.props.metadata!} />
        <FlexWrapper>
          <FlexItem>{display}</FlexItem>
          <Toolbar
            dimensions={dimensions}
            setGrid={this.setGrid}
            setView={this.setView}
            currentView={view}
          />
        </FlexWrapper>
      </div>
    );
  }
}

export default DataExplorer;
