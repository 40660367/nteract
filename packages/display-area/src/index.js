// @flow
import Display from "./display";
import RichestMime from "./richest-mime";
import Output from "./output";

import { StreamOutputComponent } from "./outputs/stream";
import { ErrorOutputComponent } from "./outputs/error";
import { ExecuteResultOutputComponent } from "./outputs/execute-result";
import { DisplayDataComponent } from "./outputs/display-data";

// export * from "./outputs";

export {
  Display,
  RichestMime,
  Output,
  StreamOutputComponent,
  ErrorOutputComponent,
  ExecuteResultOutputComponent,
  DisplayDataComponent
};
