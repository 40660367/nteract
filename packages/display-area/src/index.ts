import Display from "./display";
import Output from "./output";
import RichestMime from "./richest-mime";

export { Display, RichestMime, Output };
