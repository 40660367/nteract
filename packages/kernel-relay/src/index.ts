import { ApolloServer, gql } from "apollo-server";

import { findAll, launchKernel, Kernel } from "@nteract/fs-kernels";

const Types = gql`
  type KernelSpec {
    id: ID!
    name: String
  }

  type KernelSession {
    id: ID!
    status: String
  }
`;

const Query = gql`
  type Query {
    listKernelSpecs: [KernelSpec!]!
    running: [KernelSession!]!
  }
`;

const Mutation = gql`
  type Mutation {
    startKernel(name: String): KernelSession!
  }
`;

type StartKernel = {
  name: string;
};

const kernels: { [id: string]: Kernel } = {};

const typeDefs = [Types, Query, Mutation];
const resolvers = {
  Query: {
    listKernelSpecs: async () => {
      const kernelspecs = await findAll();

      return Object.keys(kernelspecs).map(key => {
        return { id: key, ...kernelspecs[key] };
      });
    },
    running: () => {
      return Object.keys(kernels).map(id => ({ id, status: "pretend" }));
    }
  },
  Mutation: {
    startKernel: async (_parentValue: any, args: StartKernel) => {
      /**
       *
       * const kernel
       *
       *
       */

      const kernel = await launchKernel(args.name);

      kernels[kernel.connectionInfo.key] = kernel;
      return {
        id: kernel.connectionInfo.key,
        status: "launched"
      };
    }
  }
};

async function main() {
  const mocks = {
    // By default we'll do empty objects for the JSON Scalar
    JSON: () => ({})
  };

  // In the most basic sense, the ApolloServer can be started
  // by passing type definitions (typeDefs) and the resolvers
  // responsible for fetching the data for those types.
  const server = new ApolloServer({
    typeDefs,
    resolvers: resolvers as any,
    // mocks,
    // mockEntireSchema: false,
    // Since we're playing around, enable features for introspection and playing on our current deployment
    // If this gets used in a "real" production capacity, introspection and playground should be disabled
    // based on NODE_ENV === "production"
    introspection: true,
    playground: {
      /*tabs: [
        {
          endpoint: "",
          query: ``
        }
      ]*/
    }
  });

  // This `listen` method launches a web-server.  Existing apps
  // can utilize middleware options, which we'll discuss later.
  server.listen().then(({ url }) => {
    console.log(`ðŸš€  Server ready at ${url}`);
  });
}

main();

process.on("exit", () => {
  Object.keys(kernels).map(async id => {
    await kernels[id].shutdown();
  });
});
