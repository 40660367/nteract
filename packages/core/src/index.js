//@flow
import * as actions from "./actions";
import * as actionTypes from "./actionTypes";
import * as middlewares from "./middlewares";
import * as reducers from "./reducers";
import * as themes from "./themes";
import * as epics from "./epics";
import * as selectors from "./selectors";
import * as records from "./records";

export * from "./components";
export * from "./providers";

// keeping with backwards compatiblity for now
const constants = actionTypes;

export {
  records,
  actions,
  actionTypes,
  constants,
  middlewares,
  reducers,
  themes,
  selectors,
  epics
};
