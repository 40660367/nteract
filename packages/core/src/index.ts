import * as reducers from "@nteract/reducers";
import * as epics from "@nteract/epics";
import * as selectors from "@nteract/selectors";
import * as state from "@nteract/types";
import * as actions from "@nteract/actions";

import * as middlewares from "./middlewares";

export * from "@nteract/types";

console.log("actions in core import", actions);

export { actions, middlewares, reducers, selectors, epics, state };
