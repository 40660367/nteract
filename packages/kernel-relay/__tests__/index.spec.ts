import { gql } from "apollo-server";
import { createTestClient } from "apollo-server-testing";
import { server } from "../src";

describe("Queries", () => {
  it("returns a list of kernelspecs", async () => {
    const { query } = createTestClient(server);
    const kernelspecsQuery = gql`
      query GetKernels {
        listKernelSpecs {
          name
        }
      }
    `;
    const response = await query({ query: kernelspecsQuery });
    expect(response).not.toBeNull();
    expect(response.data.listKernelSpecs.length).toBeGreaterThan(0);
  });
});
