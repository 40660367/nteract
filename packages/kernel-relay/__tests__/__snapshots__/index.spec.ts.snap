// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mutations launches a kernel 1`] = `
Object {
  "data": Object {
    "startKernel": Object {
      "id": "23431ca1-1185-42f2-9ffc-33fd1cb01db2",
      "status": "launched",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Mutations shuts down a kernel 1`] = `
Object {
  "data": Object {
    "shutdownKernel": Object {
      "id": "23431ca1-1185-42f2-9ffc-33fd1cb01db2",
      "status": "shutdown",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries returns a list of kernelspecs 1`] = `
Object {
  "data": Object {
    "listKernelSpecs": Array [
      Object {
        "name": "python3",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
